docs: Add support for Hologres

TimothyDing opened the pull request:
### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
 [db engine] Add support for Hologres

Hologres is a cloud-native Hybrid Serving & Analytical Processing (HSAP) system that is seamlessly integrated with the big data ecosystem. You can use Hologres to analyze and process PB-scale data with high concurrency and low latency. With Hologres, you can use your business intelligence (BI) tools to analyze data in multiple dimensions and explore your business in real-time.

It is fully compatible with PostgreSQL 11 and integrates seamlessly with the big data ecosystem.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
<!--- Skip this if not applicable -->

### TEST PLAN
<!--- What steps should be taken to verify the changes -->

### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

TimothyDing committed:
Add support for Hologres

* [db engine] Add support for Apache Hologres

junlincc commented:
@TimothyDing Hi Timothy, thanks for the contribution. 
@ktmud Jesse, Timothy is also an alumni from ali. please meet 😄

cc @zhaoyongjie

TimothyDing commented:
> @TimothyDing Hi Timothy, thanks for the contribution.
> @ktmud Jesse, Timothy is also an alumni from alibaba. please meet 😄
> 
> cc @zhaoyongjie

@ktmud Jesse,@junlincc
Nice to meet you!

zhaoyongjie commented:
Good to see you on GitHub, Timothy.

On Mon, Jan 25, 2021 at 10:09 AM Ye Ding <notifications@github.com> wrote:

> @TimothyDing <https://github.com/TimothyDing> Hi Timothy, thanks for the
> contribution.
> @ktmud <https://github.com/ktmud> Jesse, Timothy is also an alumni from
> alibaba. please meet 😄
>
> cc @zhaoyongjie <https://github.com/zhaoyongjie>
>
> @ktmud <https://github.com/ktmud> Jesse,@junlincc
> <https://github.com/junlincc>
> Nice to meet you!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/superset/pull/12710#issuecomment-766494067>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAPMKUTKLKJJQ7GTO7WU7KLS3THFXANCNFSM4WPVOJEQ>
> .
>

TimothyDing commented:
> Good to see you on GitHub, Timothy.
> […](#)
> On Mon, Jan 25, 2021 at 10:09 AM Ye Ding ***@***.***> wrote: @TimothyDing <https://github.com/TimothyDing> Hi Timothy, thanks for the contribution. @ktmud <https://github.com/ktmud> Jesse, Timothy is also an alumni from alibaba. please meet 😄 cc @zhaoyongjie <https://github.com/zhaoyongjie> @ktmud <https://github.com/ktmud> ***@***.*** <https://github.com/junlincc> Nice to meet you! — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub <[#12710 (comment)](https://github.com/apache/superset/pull/12710#issuecomment-766494067)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAPMKUTKLKJJQ7GTO7WU7KLS3THFXANCNFSM4WPVOJEQ> .

@zhaoyongjie 
I hope I can be more active on github in the future

villebro reviewed (commented):
Thanks for the contribution @TimothyDing ! Given that Hologres is Postgres 11 compatible, I assume all time grains and other properties defined in the Postgres db engine spec are valid?

TimothyDing commented:
> Thanks for the contribution @TimothyDing ! Given that Hologres is Postgres 11 compatible, I assume all time grains and other properties defined in the Postgres db engine spec are valid?

@villebro Nice to meet you!
Do you mean whether hologres supports PostgreSQL window functions? Could you please give me some details

villebro commented:
> > Thanks for the contribution @TimothyDing ! Given that Hologres is Postgres 11 compatible, I assume all time grains and other properties defined in the Postgres db engine spec are valid?

To be fully supported, the [time grains](https://github.com/apache/superset/blob/master/superset/db_engine_specs/postgres.py#L48-L58), [epoch conversion function](https://github.com/apache/superset/blob/master/superset/db_engine_specs/postgres.py#L69-L71) and [datetime conversion functions](https://github.com/apache/superset/blob/master/superset/db_engine_specs/postgres.py#L115-L122) should preferably be identical to the base Postgres dialect, otherwise Superset might not function properly.

TimothyDing commented:
> > > Thanks for the contribution @TimothyDing ! Given that Hologres is Postgres 11 compatible, I assume all time grains and other properties defined in the Postgres db engine spec are valid?
> 
> To be fully supported, the [time grains](https://github.com/apache/superset/blob/master/superset/db_engine_specs/postgres.py#L48-L58), [epoch conversion function](https://github.com/apache/superset/blob/master/superset/db_engine_specs/postgres.py#L69-L71) and [datetime conversion functions](https://github.com/apache/superset/blob/master/superset/db_engine_specs/postgres.py#L115-L122) should preferably be identical to the base Postgres dialect, otherwise Superset might not function properly.



> > > Thanks for the contribution @TimothyDing ! Given that Hologres is Postgres 11 compatible, I assume all time grains and other properties defined in the Postgres db engine spec are valid?
> 
> To be fully supported, the [time grains](https://github.com/apache/superset/blob/master/superset/db_engine_specs/postgres.py#L48-L58), [epoch conversion function](https://github.com/apache/superset/blob/master/superset/db_engine_specs/postgres.py#L69-L71) and [datetime conversion functions](https://github.com/apache/superset/blob/master/superset/db_engine_specs/postgres.py#L115-L122) should preferably be identical to the base Postgres dialect, otherwise Superset might not function properly.

Yes, we support it.
For example
![image](https://user-images.githubusercontent.com/47961320/105805949-feab9180-5fdd-11eb-963f-4a8883dd8c0a.png)
![image](https://user-images.githubusercontent.com/47961320/105806078-416d6980-5fde-11eb-86fd-299ac2f31066.png)
![image](https://user-images.githubusercontent.com/47961320/105806416-eb4cf600-5fde-11eb-809b-228e96a00c31.png)

villebro commented:
@TimothyDing perfect, then Hologres seems to be fully supported 👍

TimothyDing commented:
> @TimothyDing perfect, then Hologres seems to be fully supported 👍

@villebro 
I hope we could merge this pr as soon as possible.
And Hologres is a very amazing DataWarehouse.

villebro reviewed (approved):
LGTM; @srinify before I hit merge, do the docs updates look ok to you?

villebro commented:
@TimothyDing can you rebase? I think this is ready to to, so I'll be happy to merge once the conflict is resolved.

srinify commented:
@TimothyDing thank you for contributing this and welcome to the community!

I tested documentation locally and the logo rendered just fine. This is optional, but can you remove the trailing `,` in the last element of `docs/src/resources/data.js`. This still seems to work, but it's technically not valid JavaScript! (Only the first to `n-1` elements in a list object in Javascript should have a trailing comma `,`, but not the `n` element!)

![image](https://user-images.githubusercontent.com/801507/106157582-de501400-6150-11eb-9c6a-d012e1de28b7.png)

TimothyDing commented:
> @TimothyDing thank you for contributing this and welcome to the community!
> 
> I tested documentation locally and the logo rendered just fine. This is optional, but can you remove the trailing `,` in the last element of `docs/src/resources/data.js`. This still seems to work, but it's technically not valid JavaScript! (Only the first to `n-1` elements in a list object in Javascript should have a trailing comma `,`, but not the `n` element!)
> 
> ![image](https://user-images.githubusercontent.com/801507/106157582-de501400-6150-11eb-9c6a-d012e1de28b7.png)

@srinify Nice to meet. Thank you for your message, I will fix it.

srinify commented:
> > @TimothyDing thank you for contributing this and welcome to the community!
> > I tested documentation locally and the logo rendered just fine. This is optional, but can you remove the trailing `,` in the last element of `docs/src/resources/data.js`. This still seems to work, but it's technically not valid JavaScript! (Only the first to `n-1` elements in a list object in Javascript should have a trailing comma `,`, but not the `n` element!)
> > ![image](https://user-images.githubusercontent.com/801507/106157582-de501400-6150-11eb-9c6a-d012e1de28b7.png)
> 
> @srinify Nice to meet. Thank you for your message, I will fix it.

Excellent! Sorry for the confusion, I changed the screenshot I was referencing. Dang clipboard! 

Also, just a reminder to rebase off of master to fix the merge conflict!

TimothyDing committed:
Update data.js

TimothyDing committed:
Merge branch 'master' into hologres_database

TimothyDing committed:
Update data.js

TimothyDing commented:
> > > @TimothyDing thank you for contributing this and welcome to the community!
> > > I tested documentation locally and the logo rendered just fine. This is optional, but can you remove the trailing `,` in the last element of `docs/src/resources/data.js`. This still seems to work, but it's technically not valid JavaScript! (Only the first to `n-1` elements in a list object in Javascript should have a trailing comma `,`, but not the `n` element!)
> > > ![image](https://user-images.githubusercontent.com/801507/106157582-de501400-6150-11eb-9c6a-d012e1de28b7.png)
> > 
> > 
> > @srinify Nice to meet. Thank you for your message, I will fix it.
> 
> Excellent! Sorry for the confusion, I changed the screenshot I was referencing. Dang clipboard!
> 
> Also, just a reminder to rebase off of master to fix the merge conflict!



> > > @TimothyDing thank you for contributing this and welcome to the community!
> > > I tested documentation locally and the logo rendered just fine. This is optional, but can you remove the trailing `,` in the last element of `docs/src/resources/data.js`. This still seems to work, but it's technically not valid JavaScript! (Only the first to `n-1` elements in a list object in Javascript should have a trailing comma `,`, but not the `n` element!)
> > > ![image](https://user-images.githubusercontent.com/801507/106157582-de501400-6150-11eb-9c6a-d012e1de28b7.png)
> > 
> > 
> > @srinify Nice to meet. Thank you for your message, I will fix it.
> 
> Excellent! Sorry for the confusion, I changed the screenshot I was referencing. Dang clipboard!
> 
> Also, just a reminder to rebase off of master to fix the merge conflict!

@srinify 
I found that if I delete the comma in the last line, the manual will fail to build. So I added a comma again, and now all the checks have passed. 

@villebro  Please help me merge, thank you.

nytai commented:
@srinify trailing comma is valid javascript and we have a lint setting to enforce it. It helps with git diffs/blame/merge conflicts, as developers don't need to alter the line (just to add a `,`) when adding to the array.
